### PR TITLE
Nested Groups

### PR DESCRIPTION
**Pull request by BindyaBhaskar**
Modified the groups to support nested groups

### CODE REVIEW

File: esper/controllers/enterprise/group.py [hunk 1]
```diff
@@ -426,37 +535,52 @@ def add(self):
...
             if group is None or group.get('name') is None:
                 self.app.log.debug('[group-add] There is no active group.')
-                self.app.render('There is no active group.')
+                self.app.render('There is no active group. \n')
                 return
 
             group_id = group.get('id')
 
         devices = self.app.pargs.devices
         if not devices or len(devices) == 0:
```
Comment (abdulMuneerM) [inline esper/controllers/enterprise/group.py:581]: validate size of the list exceeds 1000

Comment (BindyaBhaskar) [reply esper/controllers/enterprise/group.py:581]: Will do.

File: README.md [hunk 2]
```diff
@@ -403,109 +422,133 @@ $ espercli group unset-active
...
+name            TestB
+parent_id       bb6be630-a2fc-4b57-bafa-fdda92433684
+device_count    0
+path            All devices/TestB/TestBB
+children_count  0
 ```
 
 #### 7. delete
-Remove particular group.
+Remove particular group. Here, if `group-name` is given, the first group from the response list with the given name will be removed. If `--groupid` or `-gid` option and `group-name` is given, the corresponding group will be removed.
```
[the remark this answers is not in the document]
Comment (BindyaBhaskar) [reply README.md:480]: Will change.

File: README.md [hunk 3]
```diff
@@ -515,16 +558,45 @@ $ espercli group devices [OPTIONS] [group-name]
...
+$ espercli group devices -id bb6be630-a2fc-4b57-bafa-fdda92433684 -g TestBB
+Number of Devices: 1
+ID                                    NAME           MODEL    CURRENT STATE    TAGS
+babc9cf5-2dbb-4382-bb9d-d6245941db35  DEV-ELOP-UULA  vivo     INACTIVE
+```
 
-ID                                    NAME          MODEL     CURRENT STATE  TAGS
-3ebc3afd-249b-4f10-8561-fa1a9ddb1bb7  SNA-SNL-KX37  Shoonya   ACTIVE
-c8efa083-f325-4e3b-8d20-71b7a2927ffb  SNA-SNL-3606  QUALCOMM  INACTIVE       kiosk
+#### 11. move
```
[the remark this answers is not in the document]
Comment (BindyaBhaskar) [reply README.md:572]: For the rest, the description is before options.

File: README.md [hunk 4]
```diff
@@ -344,54 +344,73 @@ $ espercli group list [OPTIONS]
...
 Use the `--active` or `-a` flag to mark this group as the active group. This will allow you to call further group commands without specifying the group.
 ```sh
 $ espercli group show [OPTIONS] [group-name]
 ```
 ##### Options
 | Name, shorthand | Default| Description|
 | -------------   |:------:|:----------|
-| --active, -a    |        | Set device as active for further device specific commands |
+| --groupid, -id  |        | Group id  |
+| --active, -a    |        | Set group as active for further device specific commands |
```
Comment (abdulMuneerM) [inline README.md:364]: Set group as active for further device specific commands => Set group as active for further group specific commands, right?

File: README.md [hunk 5]
```diff
@@ -403,109 +422,133 @@ $ espercli group unset-active
...
+parent_id       685484fa-eb6a-4ef9-a0f1-63f6febf7ce3
+device_count    1
+path            All devices/TestBB
+children_count  1
 ```
 
 #### 9. remove
-Remove devices from a group, active group is used to add devices if `--group` or `-g` option is not given explicitly.
+Remove devices from a group, active group is used to add devices if `--group` or `-g` option is not given explicitly. A maximum of 1000 devices can be removed at a time.
+Here, if `--group` or `-g` is given, devices will be removed from the first group from the response list with the given name. If `--groupid` or `-id` is also given, then devices will be removed from the corresponding group.
```
Comment (abdulMuneerM) [inline README.md:524]: Pls add like when devices are removing from a group, those will be added to an immediate parent group

File: README.md [hunk 6]
```diff
@@ -515,16 +558,45 @@ $ espercli group devices [OPTIONS] [group-name]
...
+```
 
-ID                                    NAME          MODEL     CURRENT STATE  TAGS
-3ebc3afd-249b-4f10-8561-fa1a9ddb1bb7  SNA-SNL-KX37  Shoonya   ACTIVE
-c8efa083-f325-4e3b-8d20-71b7a2927ffb  SNA-SNL-3606  QUALCOMM  INACTIVE       kiosk
+#### 11. move
+Move a group. 
+Here, if `--group` or `-g` is given, the first group from the response list with the given name will be moved. If `--groupid` or `-id` is also given, then the corresponding group will be moved. If `--parent` or `-p` is given, the first group from the response list with the given name will be the parent group. If `--parentid` or `-pid` is also given, then the corresponding group will be the parent group.
+```sh
+$ espercli group devices [OPTIONS] [group-name]
```
Comment (abdulMuneerM) [inline README.md:576]: not devices, it's move right?

File: README.md [hunk 7]
```diff
@@ -515,16 +558,45 @@ $ espercli group devices [OPTIONS] [group-name]
...
+```
+##### Options
+| Name, shorthand | Default| Description|
+| -------------   |:------:|:----------|
+| --limit, -l     |20      | Number of results to return per page |
```
Comment (abdulMuneerM) [inline README.md:581]: --limit, --offset are not needed for moving a group

**Review by abdulMuneerM — APPROVED** [review]

**Pull request merged**